pkg/zfs/volume.go

```diff
@@ -70,19 +71,45 @@ var (
 )
 
 func init() {
+	var err error
 
 	OpenEBSNamespace = os.Getenv(OpenEBSNamespaceKey)
-	if OpenEBSNamespace == "" && os.Getenv("OPENEBS_NODE_DRIVER") != "" {
-		klog.Fatalf("OPENEBS_NAMESPACE environment variable not set")
-	}
-	NodeID = os.Getenv("OPENEBS_NODE_ID")
-	if NodeID == "" && os.Getenv("OPENEBS_NODE_DRIVER") != "" {
-		klog.Fatalf("NodeID environment variable not set")
+
+	if os.Getenv("OPENEBS_NODE_DRIVER") != "" {
+		if OpenEBSNamespace == "" {
+			klog.Fatalf("OPENEBS_NAMESPACE environment variable not set for daemonset")
+		}
+		nodename := os.Getenv("OPENEBS_NODE_NAME")
+		if nodename == "" {
+			klog.Fatalf("OPENEBS_NODE_NAME environment variable not set")
+		}
+		if NodeID, err = GetNodeID(nodename); err != nil {
+			klog.Fatalf("GetNodeID failed for node=%s err: %s", nodename, err.Error())
+		}
+		klog.Infof("zfs: node(%s) has node affinity %s=%s", nodename, ZFSTopologyKey, NodeID)
+	} else if os.Getenv("OPENEBS_CONTROLLER_DRIVER") != "" {
```

Comment:
```suggestion
	} else if os.Getenv("OPENEBS_NAMESPACE") != "" {
```

Reply:
There is another if below it

Reply:
@akhilerm, we need to check for namespace only if is a controller.